lib/hd/mnemonic.js

```diff
@@ -259,7 +259,12 @@ class Mnemonic extends bio.Struct {
     this.bits = bits;
     this.language = lang;
     this.entropy = entropy;
-    this.phrase = phrase;
+
+    // Japanese likes double-width spaces.
+    if (this.language === 'japanese')
+      this.phrase = words.join('\u3000');
+    else
+      this.phrase = words.join(' ');
```

Comment:
I'm afraid this is going to cause the same problem over again (where different valid strings produce different seeds)

BIP39 states:
> To create a binary seed from the mnemonic, we use the PBKDF2 function with a mnemonic sentence (in UTF-8 NFKD) used as the password and the string "mnemonic" + passphrase (again in UTF-8 NFKD) used as the salt. 

So, I need you to check this: does this NFKD encoding specify anything about spaces?

Reply:
Since this \u3000 thing is also present in `getPhrase()` we should make sure its really the standard, otherwise Japanese phrases from hsd will be incompatible with other software

Reply:
Doing some research on this:
[Electrum](https://github.com/spesmilo/electrum/blob/0fca35fa4068ea7c9da60cd5330b85c2ca1d0398/electrum/mnemonic.py#L173) only uses single space for this. Although Electrum uses an alteration of bip39 anyway...

Reply:
Trezor (the reference implementation in BIP32, does the same thing that hsd does (japanese double space, single space for others)
https://github.com/trezor/python-mnemonic/blob/master/src/mnemonic/mnemonic.py#L70

Reply:
hm ok

Reply:
> Since this \u3000 thing is also present in `getPhrase()` we should make sure its really the standard, otherwise Japanese phrases from hsd will be incompatible with other software

I checked this a few days ago and even made a comment (now deleted), when deriving the seed phrase there is a step where it does string.normalize('nfkd'), which converts `\u3000` to ASCII. I checked with other wallet implementations too, and got the correct xpriv.